src/resources/model.js

```diff
@@ -67,17 +74,33 @@ Object.assign(pc, function () {
          * @returns {pc.Model} The loaded model.
          */
         open: function (url, data) {
+            var model = null, onLoad = function (m) {
+                model = m;
+            };
             for (var i = 0; i < this._parsers.length; i++) {
                 var p = this._parsers[i];
 
                 if (p.decider(url, data)) {
-                    return p.parser.parse(data);
+                    var result = p.parser.parse(data, onLoad);
+                    return model ? model : result;
                 }
             }
             logWARNING(pc.string.format("No model parser found for: {0}", url));
             return null;
         },
 
+        openAsync: function (url, data, asset, onLoaded, onFailed) {

```

Comment:
Because of error handling logic complexity (code might throw, or generate error and then inside another async call the same might happen) I've decided not to do that. This could be much easier to achieve if we can drop sync parsing.